src/zope/publisher/tests/test_browserrequest.py

```diff
@@ -220,6 +227,38 @@ def testFileUploadPost(self):
         # Test that we can actually read the file data
         self.assertEqual(request.form['upload'].read(), b'Some data')
 
+    def testFileUploadPost(self):
+        """Produce a Fieldstorage with a file handle that exposes
+        its filename."""
```

Comment:
@mgedmin Is this still the case? I Remember the old days where doctrings in test functions were bad. But I do not know any alive test runner which renders the docstrings instead of the test names.